### PR TITLE
Add retries to resource deletions in testing framework

### LINKED ISSUE #55860: flake: test failures with "read: connection reset by peer; some request body already written" when creating/changing objects
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:

Example message:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scalability/load.go:288
creating { ReplicationController} load-small-223
Expected error:
    <*errors.errorString | 0xc421a485e0>: {
        s: "Error creating replication controller: Post https://35.185.119.78/api/v1/namespaces/e2e-tests-load-30-nodepods-1-trc47/replicationcontrollers: read tcp 10.60.151.252:56160->35.185.119.78:443: read: connection reset by peer; some request body already written",
    }
    Error creating replication controller: Post https://35.185.119.78/api/v1/namespaces/e2e-tests-load-30-nodepods-1-trc47/replicationcontrollers: read tcp 10.60.151.252:56160->35.185.119.78:443: read: connection reset by peer; some request body already written
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scalability/load.go:540
```

It happened a few tests, not all of them scalability:
 - https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability-stable1/191
 - https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-soak-gce-2-test/3501
 - https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/48
 - https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-staging-default-latest-upgrade-cluster/79
 - https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-correctness/30
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-large-performance/25
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/77
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/78

It also happens on the GET alone, examples:
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/9011
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/9000
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/9127
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/99
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/10054
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-large-performance/45
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/92
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/93

**What you expected to happen**:

This should be retried at some level.

### PR DESCRIPTION
**Pull request by shyamjvs**
Fix https://github.com/kubernetes/kubernetes/issues/55860
(last piece towards fixing it hopefully)

/cc @wojtek-t 

```release-note
NONE
```

### CODE REVIEW

File: test/utils/delete_resources.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	}
+}
+
+func getReaperForKind(c internalclientset.Interface, kind schema.GroupKind) (kubectl.Reaper, error) {
+	return kubectl.ReaperFor(kind, c)
+}
+
+func DeletePodWithRetries(c clientset.Interface, namespace, name string, options *metav1.DeleteOptions) error {
+	deleteFunc := func() (bool, error) {
+		err := c.CoreV1().Pods(namespace).Delete(name, options)
```
Comment (wojtek-t) [inline test/utils/delete_resources.go]: Can you add Pod (and Secret and ConfigMap) to deleteResource function above, and then get rid of those dedicated functions?

Comment (shyamjvs) [reply test/utils/delete_resources.go]: Cool idea! I wasn't very sure earlier if it works, but seems so. Changed it.

**Comment by shyamjvs** [comment]
Adding approved-for-milestone as this is just a testing framework change (to save some effort for @jdumars - hope that's fine :).

**Comment by shyamjvs** [comment]
I fixed a compilation bug caused by another PR that got merged today. Re-lgtm'ing based on above.

**Pull request merged**